src/h5web/visualizations/heatmap/AxisGrid.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from 'react';
+import { Dom, useThree, useFrame } from 'react-three-fiber';
+import { scaleLinear } from 'd3-scale';
+import IndexAxis from './IndexAxis';
+import styles from './HeatmapVis.module.css';
+
+interface AxisDomains {
+  left: [number, number];
+  bottom: [number, number];
+}
+
+interface Props {
+  dims: [number, number];
+  axisOffsets: [number, number];
+}
+
+function AxisGrid(props: Props): JSX.Element {
+  const { dims, axisOffsets } = props;
+  const [rows, cols] = dims;
+  const [leftAxisWidth, bottomAxisHeight] = axisOffsets;
+
+  const { camera, size } = useThree();
+  const { width, height } = size;
+
+  const [domains, setDomains] = useState<AxisDomains>({
+    left: [0, rows],
+    bottom: [0, cols],
+  });
+
+  useFrame(() => {
+    const { position, zoom } = camera;
+    const factor = 1 / zoom;
+
+    const leftScale = scaleLinear()
+      .domain([-height / 2 + bottomAxisHeight, height / 2])
+      .range([0, rows]);
+    const bottomScale = scaleLinear()
+      .domain([-width / 2 + leftAxisWidth, width / 2])
+      .range([0, cols]);
```

Comment:
I have done here with d3's `scale` but I can manage to implement it mathematically if we think it may improve the performance